Sources/Sentry/Public/SentrySpan.h

```diff
@@ -1,9 +1,97 @@
-#import <Foundation/Foundation.h>
+#import "SentrySpanContext.h"
+#import <Sentry/Sentry.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class SentryTransaction, SentrySpanId, SentryId, SentryHub;
+
 NS_SWIFT_NAME(Span)
-@protocol SentrySpan <NSObject>
+@interface SentrySpan : SentrySpanContext
```

Comment:
L: on Java, a Span has the instance of the `Hub` so we could associate an error to a span
See: https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Span.java#L65-L67

.NET also does not do it, yet, so it'd be fine to skip it for now, but keep in mind that we might want to do it at a later point